src/assistants/openai/openai_api_version.rs

```diff
@@ -0,0 +1,175 @@
+use anyhow::{anyhow, Result};
+use reqwest::header::{self, HeaderMap, HeaderValue};
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+use std::str::FromStr;
+
+use crate::constants::OPENAI_API_URL;
+
+#[derive(Deserialize, Serialize, Debug, Clone, Eq, PartialEq)]
+pub enum OpenAIAssistantVersion {
+    V1,
+    V2,
+    Azure,
+}
+
+impl OpenAIAssistantVersion {
+    pub(crate) fn get_endpoint(&self, resource: &OpenAIAssistantResource) -> String {
+        //OpenAI documentation: https://platform.openai.com/docs/models/model-endpoint-compatibility
+        let base_url = match self {
+            OpenAIAssistantVersion::V1 | OpenAIAssistantVersion::V2 => {
+                format!("{OPENAI_API_URL}/v1", OPENAI_API_URL = *OPENAI_API_URL)
+            }
+            OpenAIAssistantVersion::Azure => {
+                format!("{OPENAI_API_URL}/openai", OPENAI_API_URL = *OPENAI_API_URL)
+            }
+        };
+
+        let path = match resource {
+            OpenAIAssistantResource::Assistants => format!("{base_url}/assistants"),
+            OpenAIAssistantResource::Assistant { assistant_id } => {
+                format!("{base_url}/assistants/{assistant_id}")
+            }
+            OpenAIAssistantResource::Threads => format!("{base_url}/threads"),
+            OpenAIAssistantResource::Messages { thread_id } => {
+                format!("{base_url}/threads/{thread_id}/messages")
+            }
+            OpenAIAssistantResource::Runs { thread_id } => {
+                format!("{base_url}/threads/{thread_id}/runs")
+            }
+            OpenAIAssistantResource::Run { thread_id, run_id } => {
+                format!("{base_url}/threads/{thread_id}/runs/{run_id}")
+            }
+            OpenAIAssistantResource::Files => format!("{base_url}/files"),
+            OpenAIAssistantResource::File { file_id } => format!("{base_url}/files/{file_id}"),
+            OpenAIAssistantResource::VectorStores => format!("{base_url}/vector_stores"),
+            OpenAIAssistantResource::VectorStore { vector_store_id } => {
+                format!("{base_url}/vector_stores/{vector_store_id}")
+            }
+            OpenAIAssistantResource::VectorStoreFileBatches { vector_store_id } => {
+                format!("{base_url}/vector_stores/{vector_store_id}/file_batches")
+            }
+        };
+
+        // Add suffix if needed
+        match self {
+            OpenAIAssistantVersion::Azure => format!("{path}?api-version=2024-05-01-preview"),
+            _ => path,
+        }
+    }
+
+    pub(crate) fn get_headers(&self, api_key: &str) -> HeaderMap {
```

Comment:
These next methods were not modified. I just moved this enum to a separate file for easier code management.